pkg/mqtrigger/scalermanager.go

```diff
@@ -0,0 +1,265 @@
+				}
+			}()
+		},
+	})
+	controller.Run(context.Background().Done())
+	return nil
+}
+
+func getScaledObject(mqt *fv1.MessageQueueTrigger) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "ScaledObject",
+			"apiVersion": "keda.k8s.io/v1alpha1",
+			"metadata": map[string]interface{}{
+				"name":      mqt.ObjectMeta.Name,
+				"namespace": mqt.ObjectMeta.Namespace,
+			},
+			"spec": map[string]interface{}{
+				"cooldownPeriod":  &mqt.Spec.CooldownPeriod,
+				"maxReplicaCount": &mqt.Spec.MaxReplicaCount,
+				"minReplicaCount": &mqt.Spec.MinReplicaCount,
+				"pollingInterval": &mqt.Spec.PollingInterval,
+				"scaleTargetRef": map[string]interface{}{
+					"deploymentName": mqt.ObjectMeta.Name,
+				},
+				"triggers": []interface{}{
+					map[string]interface{}{
+						"type":     mqt.ObjectMeta.Name,
+						"metadata": mqt.Spec.Metadata,
+						"authdata": mqt.Spec.Authdata,
+					},
+				},
+			},
+		},
+	}
+}
+
+func getDeploymentSpec(mqt *fv1.MessageQueueTrigger, routerUrl string) *appsv1.Deployment {
+	url := routerUrl + "/" + strings.TrimPrefix(utils.UrlForFunction(mqt.Spec.FunctionReference.Name, mqt.ObjectMeta.Namespace), "/")
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: mqt.ObjectMeta.Name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": mqt.ObjectMeta.Name,
+				},
+			},
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": mqt.ObjectMeta.Name,
+					},
+				},
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:            mqt.ObjectMeta.Name,
+							Image:           "rahulbhati/test:3",
+							ImagePullPolicy: "Always",
+							Env: []apiv1.EnvVar{
+								{
+									Name:  "BROKER_LIST",
+									Value: mqt.Spec.Metadata["brokerList"],
+								},
+								{
+									Name:  "BOOTSTRAP_SERVERS",
+									Value: mqt.Spec.Metadata["bootstrapServers"],
+								},
+								{
+									Name:  "CONSUMER_GROUP",
+									Value: mqt.Spec.Metadata["consumerGroup"],
+								},
+								{
+									Name:  "TOPIC",
+									Value: mqt.Spec.Topic,
+								},
+								{
+									Name:  "LAG_THRESHOLD",
+									Value: mqt.Spec.Metadata["lagThreshold"],
+								},
+								{
+									Name:  "AUTH_MODE",
+									Value: mqt.Spec.Metadata["authMode"],
+								},
+								{
+									Name:  "USERNAME",
+									Value: mqt.Spec.Metadata["username"],
+								},
+								{
+									Name:  "PASSWORD",
+									Value: mqt.Spec.Metadata["password"],
+								},
+								{
+									Name:  "CA",
+									Value: mqt.Spec.Metadata["ca"],
+								},
+								{
+									Name:  "CERT",
+									Value: mqt.Spec.Metadata["cert"],
+								},
+								{
+									Name:  "KEY",
+									Value: mqt.Spec.Metadata["key"],
+								},
```

Comment:
All these are right now specific to Kafka - and hence not applicable for others. We should have all the environment variables populated by reading from Metadata keys and capitalizing them (Or some similar scheme). So that this is generic irrespective of the trigger.

Reply:
Done